Feat/code editor rtc

RTC-13 opened the pull request:
Coming soon

RTC-13 committed:
Feat-CodeEditor-Start

Implementation of Code Editor

RTC-13 committed:
Merge pull request #923 from RTC-13/feat/code-editor

Feat-CodeEditor-Start

RTC-13 committed:
Feat-CodeEditor-InputHandling

Added user input handling for Code Editor

RTC-13 committed:
Merge pull request #931 from RTC-13/feat/code-editor-rtc

Feat-CodeEditor-InputHandling

jschanker committed:
Add generated `python-worker` script from `react-py`

jschanker committed:
Updated `react-py` `python-worker script`

jschanker committed:
Added `react-py` `python-console-worker` script

jschanker committed:
Merge pull request #933 from navgurukul/feat/code-editor-rtc-worker

Feat/code editor - Added worker scripts from `react-py`

RTC-13 committed:
Feat/React-py added Python code execution

RTC-13 committed:
Merge pull request #934 from RTC-13/feat/code-editor-rtc

Feat/React-py added Python code execution

RTC-13 committed:
Removed Vizualize Button

RTC-13 committed:
Remove createVisulizeURL

RTC-13 committed:
Reformatting + Run Button fail safe

RTC-13 committed:
Styling PythonEditor.js

RTC-13 committed:
Code Editor Styling

Initial commit for Code Editor Styling

RTC-13 committed:
Reset Code Button functionality + Fixed Output overflowing

RTC-13 committed:
Removed unnecessary code

RTC-13 committed:
Update PythonEditor.js

RTC-13 committed:
Update PythonEditor.js

Conditional rendering

RTC-13 committed:
Added the ability to disable ability to run and edit code editor

RTC-13 committed:
Refactored Code Editor

Fixes / Refactoring Code Editor

RTC-13 committed:
fix:CodeEditor readOnly

RTC-13 committed:
Remove unused hook import

RTC-13 committed:
Cursor offset

possible solution

RTC-13 committed:
Cursor offset fix

jschanker committed:
fix: don't crash on editing (use correct input for setEditorState)

RTC-13 committed:
fix-input for run button, fix-input for reset button

RTC-13 committed:
Update PythonEditor.js

reformat

jschanker reviewed (commented):
Some initial comments

RTC-13 committed:
Update src/components/CodeEditor/CodeMirror.js

Co-authored-by: jschanker <jschanker@molloy.edu>

RTC-13 committed:
Indentation constant

RTC-13 committed:
Update src/components/CodeEditor/PythonEditor.js

Remove console.log

Co-authored-by: jschanker <jschanker@molloy.edu>

RTC-13 committed:
Update src/components/CodeEditor/PythonEditor.js

Co-authored-by: jschanker <jschanker@molloy.edu>

RTC-13 committed:
Update PythonEditor.js

Replaced border color with theme

RTC-13 committed:
Merge branch 'feat/code-editor-rtc' of https://github.com/navgurukul/bhanwari-devi into feat/code-editor-rtc

RTC-13 committed:
Code Review updates, Themes, updated output

RTC-13 committed:
Update PythonEditor.js

Removed unused useEffect

RTC-13 committed:
Added JSDoc

RTC-13 committed:
Update PythonEditor.js

fixed crash

RTC-13 committed:
Fixed editor state reset (Changed Styling)

RTC-13 committed:
Added Code Editor Constants

jschanker committed:
fix: PathwayExercise-Remove useless styling, update theme breakpoints

jschanker committed:
small styling change to Next arrow, use Box instead of div

jschanker committed:
Merge branch 'main' into feat/code-editor-rtc

jschanker committed:
fix: multiline comment during merge

jschanker reviewed (dismissed):
Future TODO:
1. Currently every code editor will cause a new request to load `Pyodide`. This is inefficient, but should not cause any issues. For the future, we could fix this by moving the `usePython` hook up to an ancestor and passing down the relevant properties so all code editors can run their code without triggering their own requests to load Pyodide.
2. The workaround we used to get `react-py` to work is causing a number of tests to fail: https://github.com/navgurukul/bhanwari-devi/pull/924#issuecomment-1493604832 . We might be able to reduce the number of these errors if we had copied the code as is instead of running it through Babel to transpile to ES2015. This should be fixed once we update packages.

komalahire committed:
remove console.log

jschanker committed:
feat: Quick hack to allow input at beginning of Python code

jschanker committed:
fix: Evaluate all initial lines of Python code with input at once in JS

jschanker committed:
fix: Allow blank lines in Python code between `input`s

RTC-13 committed:
Fix: error caused when there was no input function used in the code editor

Fixed error which happened when there was no user use of "input" activating the alert. Fixed by checking the if the noInputIndex was equal to zero, then we run the code normally and return out of the function.

RTC-13 committed:
Remove console.log

jschanker committed:
fix: Python code editor - allow comments on input lines, no replacements in literals

jschanker committed:
fix: Allow string literals with quotes for `input` lines in Python code editor

jschanker committed:
Merge pull request #1026 from navgurukul/feat/code-editor-rtc-input-hack

feat: Quick hack to allow input at beginning of Python code

jschanker committed:
fix: exercise one behind error - `PathwayExercise`

jschanker committed:
chore: Minor cleanup of Assessment

jschanker committed:
fix: selectively make request for completed ids - `PathwayExercise`

jschanker committed:
fix: replace removed `currentCourse` with `courseId`

jschanker committed:
fix: Check previous (not current) exercise id for making request

jschanker committed:
fix: first time `useEffect` `course[previousExercise]` will be `undefined`

jschanker committed:
fix: Mark exercise as complete after response

komalahire committed:
Merge pull request #1106 from navgurukul/fix/one-behind-error-pathway-exercise

Fix/one behind error pathway exercise

Tamannatanwa committed:
This commit adds new fields for student login via username and password on the login page.

amansharmma committed:
Updated the Partner Page to add report genration.

Tamannatanwa committed:
implemented Poonam's feedback and separated the login by Gmail and login by username and password codes

amansharmma committed:
Updated the assessment FUnctionality

Tamannatanwa committed:
recreated the table by making some modifications to the existing one, adding a few columns and removing others.

saloni-080601 committed:
work on video skiping part for peepula india

komalahire committed:
Merge pull request #1111 from navgurukul/Peepul-india-partner-report

Updated the Partner Page to add report generation.

Tamannatanwa committed:
add margin for small screen

Tamannatanwa committed:
work on add student and edit student feedback

komalahire committed:
Merge pull request #1108 from navgurukul/meraki_student_login_page

This commit adds new fields for student login via username and passwo…

Tamannatanwa committed:
Work on designing and fixing errors in adding student and editing student.

jschanker committed:
perf: Load single instance of Pyodide by moving hook up to parent

jschanker committed:
perf: Load single instance of Pyodide by referring to hook from parent

komalahire committed:
Merge pull request #1112 from navgurukul/partners_studentdata_table

recreated the table by making some modifications to the existing one,…

jschanker committed:
fix: Conditional loading of Pyodide, pass to component

jschanker committed:
fix: Load Python hook unconditionally for now

Tamannatanwa committed:
Made changes to the addStudent and login APIs to handle errors using error codes, and improved the responsiveness of the login page.

komalahire committed:
Merge pull request #1114 from navgurukul/AddStudent_Login_Api_Handling

Made changes to the addStudent and login APIs to handle errors using …

komalahire committed:
Merge pull request #1113 from navgurukul/videoSkipPArt

work on video skiping part for peepula india

jschanker committed:
Merge branch 'dev' into feat/code-editor-rtc
